SRD redshift distributions

nikosarcevic opened the pull request:
Hi, @elisachisari and @chrgeorgiou ,

As promised, I've created a branch where I added the code to generate the SRD redshift distributions (for lenses and sources for Y1 and Y10).

There is a parameter file (in parameters folder), py file with the code and an example notebook.
Let me know if you want more functionality added or you are ok with how it is.

Niko

nikosarcevic committed:
added the code that generates srd dndz

uploads contain the background code, example notebook, parameter file (stored in a folder) and output files (.csv and plots as .pdf).

nikosarcevic committed:
delete the output file (clutter)

nikosarcevic committed:
delete the output file (clutter)

nikosarcevic committed:
delete the output file (clutter)

nikosarcevic committed:
delete the output file (clutter)

nikosarcevic committed:
delete the output file (clutter)

nikosarcevic committed:
delete the output file (clutter)

nikosarcevic committed:
minor changes in the explanations and plots

nikosarcevic committed:
added a script to bin the SRD redshift distributions

code adopted from FISK pipeline (by Niko).
includes the code that does the binning (binning.py) and an example notebook.

nikosarcevic committed:
file name change

nikosarcevic committed:
delete binning notebook

forgot to run all cells, going to upload it again

nikosarcevic committed:
binning example notebook 

Jupyter notebook that showcases how to bin following the SRD recipe.
More explanation in the notebook and in binning.py file.

nikosarcevic commented:
> Hi, @elisachisari and @chrgeorgiou ,
> 
> As promised, I've created a branch where I added the code to generate the SRD redshift distributions (for lenses and sources for Y1 and Y10).
> 
> There is a parameter file (in parameters folder), py file with the code and an example notebook. Let me know if you want more functionality added or you are ok with how it is.
> 
> Niko

I have now added the binning procedure to this branch, as per @heatherprince request.

nikosarcevic committed:
added more comments to the notebook

nikosarcevic committed:
import reshuffle PEP8

nikosarcevic committed:
Update lsst_desc_parameters.yaml

removed some comments from source limiting magnitude as the values were confirmed by Jeff Newman

elisachisari commented on parameters/lsst_desc_parameters.yaml:
There is a lot here we don't seem to need for the redshift distributions. What was the motivation for including all the parameters?

nikosarcevic replied:
motivation is to have the SRD baseline params somewhere. I do plan to convert it to maybe a pdf or something else? I can remove the unnecessary params, not a problem

elisachisari replied:
Yes, I think this belongs in the forecasting repo insted

elisachisari commented on Generate the SRD redshift distributions.ipynb:
Could you explain somewhere how these were validated? Was there any significant difference with the SRD? To what tolerance do they match?

nikosarcevic replied:
Hi, Elisa

I have finally added the validation notebook https://github.com/LSSTDESC/CCLX/blob/SRD-Redshift-Distributions/SRD_binning_validation.ipynb

As explained in there, when I was constructing the pipeline, I have used some code from Danielle Leonard as inspiration (developed by her and Markus Rau). I am comparing the output of the SRD binning from FISK and the output I have from Danielle. She does not have the code online so I am not sure how to tackle this. Maybe we can ask her to confirm.

I have performed some audition tests as you will see in the notebook. 

After you give the blessing we can either keep this code and data somewhere or we can remove it and I can store it on my GH repo.

Thank you for being patient with this!
Niko

elisachisari replied:
Thanks for this - I see now a validation of the source bins and n(z). What about the validation for the lens bins? Do you think we should add something?

nikosarcevic replied:
Hi, Elisa

parameters (yaml) should be here as the code is reading from it.

Re lenses -- I have nothing to compare it to but I have followed the instructions from the SRD (equidistant spacing etc). I do not know if anyone has something I can compare to?

nikosarcevic replied:
I would trust it but of course if anyone want to double check the lens bins, I would appreciate that as a sanity check! :)

nikosarcevic replied:
I have obtained a code for the SRD distributions from @chrgeorgiou and produced the lens and soruce bins for SRD Y1 and Y10. Then I have compared them to the lens and source bins from this code (adapted FISK code). As you can see in the plots attached, I am getting a 1:1 agreement.
All issues are now resolved, I believe and @elisachisari  can bless this merge.

NOTE: FISK bins are the bins from my code, DL refers to the bins from Danielle Leonard and CG stands for Christos Georgiou.
<img width="821" alt="Screenshot 2023-10-03 at 23 54 38" src="https://github.com/LSSTDESC/CCLX/assets/32773944/93c638a1-ffff-4834-8608-9241c657d78e">
<img width="827" alt="Screenshot 2023-10-03 at 23 57 46" src="https://github.com/LSSTDESC/CCLX/assets/32773944/624326c2-3154-4020-bab1-9994e45fcf67">

elisachisari replied:
Thanks for checking this too

elisachisari reviewed (changes requested):
Runs for me without issues. Some comments. 
Could you also add somewhere in the notebooks what the output files are (.csv) so people notice they are there and also what is contained there?

nikosarcevic committed:
binning modification: dict keys

I have only changed the keys of the dictionary for the bins. Before the key of each bin was the center of the bin. Now the bin dict keys are just integers which makes everything much cleaner and easier to handle later (floating point precision problems can arise if not integer).

Other than that no on the changes were implemented.

nikosarcevic committed:
added binning validation 

upon request by Elisa. Added a validation notebook with detailed comparison between the FISK binning procedure for SRD and the benchmark data from Danielle Leonard.

nikosarcevic committed:
added an example of baryon models

need to be checked by Elisa (or Tilman)

nikosarcevic committed:
Delete Baryons Modules.ipynb

nikosarcevic committed:
examples for baryonic models

needs some touchups (more info) and needs to be checked by Elisa (or Tilman)

nikosarcevic committed:
fixed typos, added more info

elisachisari reviewed (approved):
All suggested changes have been incorporated and new validation tests produced as requested
